benchmarks/native-beam/native-beam-spark/src/main/java/cz/seznam/euphoria/benchmarks/beam/StatelessJavaSerializer.java

```diff
@@ -64,8 +63,8 @@ public void write(Kryo kryo, Output output, Object object) {
 
   @SuppressWarnings("unchecked")
   public Object read (Kryo kryo, Input input, Class type) {
-    try {
-      return new ObjectInputStreamWithClassLoader(input, kryo.getClassLoader()).readObject();
+    try (ObjectInputStream objectStream = new ObjectInputStreamWithClassLoader(input, kryo.getClassLoader())) {
+      return objectStream.readObject();
     } catch (Exception e) {
       throw new KryoException("Error during Java deserialization.", e);
     }
```

Comment:
Many thanks, you're right, it would cause the Kryo `input/output` to be closed too early. I reverted this change and added explaining comment.